### PR TITLE
[NETBEANS-54] Module Review db.sql.editor

### PR DESCRIPTION
**Pull request by matthiasblaesing**
- no external library

- checked Rat report; unrecognized license headers manually changed, 
  missing headers added, ignored *.form (see central problems) and 
  SQLExample. The latter file is used in the GUI as a sample and
  in addition does not reach a level of creativity.

- skimmed through the module, did not notice additional problems

- Modified tests to allow comments in *.test and *.pass files

### CODE REVIEW

File: db.sql.editor/test/unit/src/org/netbeans/modules/db/sql/editor/completion/SelectCompletionQueryTest.java [hunk 1]
```diff
@@ -125,11 +125,11 @@ public void runTest() throws Exception {
...
         StringBuilder sqlData = new StringBuilder();
-        List<String> modelData = new ArrayList<String>();
-        BufferedReader reader = new BufferedReader(new InputStreamReader(SelectCompletionQueryTest.class.getResource(getName() + ".test").openStream(), "utf-8"));
-        try {
+        List<String> modelData = new ArrayList<>();
+        try (InputStream is = SelectCompletionQueryTest.class.getResourceAsStream(getName() + ".test");
+                BufferedReader reader = new BufferedReader(new InputStreamReader(is, "utf-8"))) {
             boolean separatorRead = false;
-            for (String line; (line = reader.readLine()) != null;) {
+            for (String line = reader.readLine(); line != null; line = reader.readLine()) {
```
Comment (emilianbold) [inline db.sql.editor/test/unit/src/org/netbeans/modules/db/sql/editor/completion/SelectCompletionQueryTest.java]: Wasn't the previous `for` line better?

File: db.sql.editor/test/unit/src/org/netbeans/modules/db/sql/editor/completion/SelectCompletionQueryTest.java [hunk 2]
```diff
@@ -143,27 +143,19 @@ public void testCompletion() throws Exception {
...
-            } finally {
-                writer.close();
             }
             File pass = new File(getWorkDir(), getName() + ".pass");
-            InputStream input = SelectCompletionQueryTest.class.getResource(getName() + ".pass").openStream();
-            try {
-                copyStream(input, pass);
-            } finally {
-                input.close();
+            try (InputStream input = SelectCompletionQueryTest.class.getResource(getName() + ".pass").openStream()) {
```
Comment (emilianbold) [inline db.sql.editor/test/unit/src/org/netbeans/modules/db/sql/editor/completion/SelectCompletionQueryTest.java]: Changing this was not necessary. But if we do it, why not `getResourceAsStream`?

File: db.sql.editor/test/unit/src/org/netbeans/modules/db/sql/editor/completion/SelectCompletionQueryTest.java (hunk 1 above)
Comment (matthiasblaesing) [reply db.sql.editor/test/unit/src/org/netbeans/modules/db/sql/editor/completion/SelectCompletionQueryTest.java]: I consider the previous one less readable. The previous version mixed initialization, loop advance and checking and these parts are now separated into their corresponding sections.

**Comment by matthiasblaesing** [comment]
Thanks for looking into this. I let the changed for loop in place and updated the usage of `getResource().openStream()` to `getResourceAsStream`.

File: db.sql.editor/test/unit/src/org/netbeans/modules/db/sql/editor/completion/SelectCompletionQueryTest.java (hunk 1 above)
Comment (emilianbold) [reply db.sql.editor/test/unit/src/org/netbeans/modules/db/sql/editor/completion/SelectCompletionQueryTest.java]: I believe the canonical trick is to have

```java
String line;
while((line = reader.readLine()) != null) {
  ...
}
```

And the original `for` just mixed those to limit the scope of the `line` variable to the `for` block... which is nice.

Your `for` repeats the `reader.readLine()` call which seems confusing.

Comment (matthiasblaesing) [reply db.sql.editor/test/unit/src/org/netbeans/modules/db/sql/editor/completion/SelectCompletionQueryTest.java]: I'll revert my change to this loop and then push to master.

**Pull request merged**